engine/src/main/java/nl/inl/blacklab/search/DocUtil.java

```diff
@@ -377,32 +299,43 @@ public List<Concordance> makeConcordancesFromContentStore(Field field, int[] sta
         return rv;
     }
 
-    @Override
-    public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        result = prime * result + id;
-        result = prime * result + ((index == null) ? 0 : index.hashCode());
-        return result;
-    }
+    /**
+     * Get part of the contents of a document.
+     *
+     * Pass -1 for start and end positions to get the whole content.
+     *
+     * @param index our indx
+     * @param id document id
+     * @param d Lucene document if available, otherwise null
+     * @param field field to get contents for
+     * @param startAtWord token position to start at
+     * @param endAtWord token position to end at (first token not in the resulting snippet)
+     * @return contents
+     */
+    static String contents(BlackLabIndex index, int id, Document d, Field field, int startAtWord, int endAtWord) {
+        if (d == null)
```

Comment:
Do we really need to do this? It seems like it would be better to either pass a `Document` or  an `id`?

Reply:
Unfortunately, we cannot get the docId from the Document, and the calling method needs both for other operations as well. We could only pass docId and retrieve the Document again, but this is wasteful if we already have it available.